libsrc/apple2/ser/a2.gs.s

```diff
@@ -314,20 +314,19 @@ SER_CLOSE:
 
 SER_OPEN:
         bit     $C082                   ; Check if this is a IIgs
-        lda     $FE1F                   ; https://prodos8.com/docs/technote/misc/07/
-        cmp     #$60                    ; Everything but the IIgs has an RTS there
-        bne     HardwareFound
+        sec
+        jsr     $FE1F                   ; https://prodos8.com/docs/technote/misc/07/
+        bcc     IIgs
 
-        ; Device (hardware) not found
         bit     $C080
```

Comment:
BIT doesn't set the carry flag. So you can put the BIT $C080 before the BCC - and thus reuire it only once. Maybe the most elegant looking code would be:
```
sec
bit $C082
jsr $FE1F
bit $c080
bcc IIgs
```